### PR TITLE
Use flex layout on context menu items

### PR DESCRIPTION
**Pull request by jevenski**
This patch does modifications to the context menu. After this patch is merged, the search menu on the top navbar can be more easily replaced with the bootstrap dropdown menu implementation, just like the plugins dropdown menu.

### Commit Message
- Split menu-item selection and sub-menu expansion icons from the background image, and use pseudo-elements for them.
- Use CSS variables on menu background and font color. This simplifies theme adaption process and reduces code duplication.
- Use inline base64 string for menu selection and expansion icons, replacing previous image urls. This saves a few requests for images and enhances performance by reducing load time.

### CODE REVIEW

**Review by stickz — CHANGES_REQUESTED** [review]
Remove `Details...` from `CMenu` in this pull request. This is useless because it simply links to the `General` tab bellow. The user can display this tab in 1 click instead of 2 clicks.

**Comment by koblack** [comment]
1. The separation line is not centered
image

2. Too much space between menu and submenu
image
image
image

File: css/style.css [hunk 1]
```diff
@@ -206,15 +209,69 @@ div#sc ul {margin: 0; padding: 0; list-style: none}
...
-ul.CMenu li a.sel {background: transparent url(../images/menusel.gif) no-repeat scroll 4px center}
-ul.CMenu li a:not(.dis):hover {background-color: var(--menu-highlight-background-color); color: var(--menu-highlight-color)}
+
+.CMenu {
+  position: absolute;
+  display: none;
+  border: 2px ridge var(--menu-border-color);
+  border-radius: 5px;
+  background: var(--menu-background-color);
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (jevenski) [reply css/style.css:219]: Above is a screenshot of `.CMenu` with `padding: 0.25rem 1rem;`. I think `padding: 0;` looks better. Plus, the menu item links have a padding of 0.25rem on them, so the texts won't collide with the menu edge.

File: css/style.css [hunk 2]
```diff
@@ -206,15 +209,69 @@ div#sc ul {margin: 0; padding: 0; list-style: none}
...
+.CMenu li hr {
+  width: 90%;
+  margin: 0;
+}
+.CMenu li a:not(.dis):hover {
+  background-color: var(--menu-highlight-background-color);
+  color: var(--menu-highlight-color);
+}
 
 ul.CMenu li:hover ul {display: block; position: absolute; top: 0; left: 146px; width:150px;}
```
[the remark this answers is not in the document]
Comment (jevenski) [reply css/style.css:276]: The submenu left offset will need to be set dynamically using JavaScript to address @koblack 's feedback.

**Comment by stickz** [comment]
@jevenski Looks good, thanks! Merged. I added 0.25rem top and bottom padding to the CMenu. The text was too close to the edge when testing. I didn't add any left and right padding, as there is already padding for that.

**Pull request merged**